nextcord/polls.py

```diff
@@ -0,0 +1,441 @@
+
+    def to_dict(self) -> PollAnswerCountPayload:
+        payload: PollAnswerCountPayload = {
+            "id": self.id,
+            "me_voted": self.me_voted,
+            "count": self.count,
+        }
+        return payload
+
+
+class PollResults:
+    """The result of the poll.
+
+    This contains the numbers of vote for each answer.
+
+    .. versionadded:: 3.0
+
+    Attributes
+    ----------
+    is_finalized: :class:`bool`
+        Whether the votes has been precisely counted.
+    answer_counts: List[:class:`PollAnswerCount`]
+        The counts for each answer.
+    poll: :class:`Poll`
+        The poll associated with this result.
+    """
+
+    __slots__ = (
+        "is_finalized",
+        "answer_counts",
+        "poll",
+    )
+
+    def __init__(self, data: PollResultsPayload, poll: Poll) -> None:
+        self.is_finalized: bool = data["is_finalized"]
+        self.answer_counts: List[PollAnswerCount] = [
+            PollAnswerCount(answer_count, poll) for answer_count in data["answer_counts"]
+        ]
+        self.poll: Poll = poll
+
+    def __repr__(self) -> str:
+        return (
+            "<PollResults "
+            f"is_finalized={self.is_finalized!r} "
+            f"answer_counts={self.answer_counts!r} "
+            ">"
+        )
+
+    def to_dict(self) -> PollResultsPayload:
+        payload: PollResultsPayload = {
+            "is_finalized": self.is_finalized,
+            "answer_counts": [answer_count.to_dict() for answer_count in self.answer_counts],
+        }
+        return payload
+
+
+class PollCreateRequest:
+    """A poll create request.
+
+    You must use this to create a poll.
+
+    .. versionadded:: 3.0
+
+    Attributes
+    ----------
+    question: Union[:class:`str`, :class:`PollMedia`]
+        The question of the poll. Currently only `text` is supported.
+        If a :class:`str` was passed, it's automatically converted to a :class:`PollMedia`.
+    duration: Optional[:class:`int`]
+        The number of hours this poll should be open for. Defaults to ``24``. Max ``768`` (32 days).
+    allow_multiselect: Optional[:class:`bool`]
+        Whether voters can select more than one choice. Defaults to False.
+    answers: Optional[List[:class:`PollAnswer`]]
+        The answers to the poll.
+        If omitted, answers must be added later with :func:`PollCreateRequest.add_answer`.
+    layout_type: Optional[:class:`PollLayoutType`]
+        The layout type of the poll. Defaults to :attr:`PollLayoutType.default`
+    """
+
+    __slots__ = (
+        "question",
+        "answers",
+        "duration",
+        "allow_multiselect",
+        "layout_type",
+    )
+
+    def __init__(
+        self,
+        question: Union[str, PollMedia],
+        duration: int = 24,
+        allow_multiselect: bool = False,
+        layout_type: Optional[PollLayoutType] = MISSING,
+        answers: Optional[List[PollAnswer]] = None,
```

Comment:
Creating a `PollAnswer` in a list of already having to make `PollAnswers` is quite tedious, here (or in a user instantiate-able `Poll`) there should be the possibility of using `str` for simple results.

```python
answers=[
    PollAnswer(poll_media=PollMedia(text="one")),
    PollAnswer(poll_media=PollMedia(text="two")),
    PollAnswer(poll_media=PollMedia(text="three")),
],
```

Reply:
I imagine they're not going to bloat up polls for a while, I think at least accepting strings in the answer list would be the simplest whilst still giving the full capability of expansive polls with the current syntax.